ocaml/xapi/xapi_vdi.ml

```diff
@@ -307,7 +307,7 @@ let update_allowed_operations_internal ~__context ~self ~sr_records ~pbd_records
   in
   let all = Db.VDI.get_record_internal ~__context ~self in
   let allowed =
-    let check x = match check_operation_error ~__context ~sr_records ~pbd_records ~vbd_records ha_enabled all self x with None ->  [ x ] | _ -> [] in
+    let check x = match check_operation_error ~__context ~sr_records ~pbd_records ?vbd_records ha_enabled all self x with None ->  [ x ] | _ -> [] in
```

Comment:
I see you pass the optional type through here, so we do not loose/overwrite the vbd records, but instead look it up from the DB.